### PR TITLE
Fixes #14634: Add a generic method to add ace for a given user on a given file

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14634

### CODE REVIEW

File: tree/30_generic_methods/permissions_user_acl_present.cf [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+#
+# @documentation The `permissions_*acl_*` manage the POSIX ACL on files and directories.
+# 
+# Please note that the mask will be automatically recalculated when editing ACLs.
+#
+# #### Parameters
+#
+# ##### Path
+#
+# Path can be a regex with the following format:
```
Comment (peckpeck) [inline tree/30_generic_methods/permissions_user_acl_present.cf]: this is globbing not regex

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR rebased

File: tests/acceptance/30_generic_methods/permissions_user_acl_present.cf [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+
+      "files[4]"     string => "file4/.";
+      "recursive[4]" string => "true";
+      "user[4]"      string => "bin";
+      "ace[4]"       string => "-rw";
+      "create[4]"    string => "true";
+      "initial[4]"    slist => { "user:*:rwx", "group:*:rx", "user:bin:-rwx", "group:bin:r", "all:rx" };
+      "exp_lines[4]"  slist => { "user::rwx", "group::r-x", "user:bin:---", "group:bin:r--", "other::r-x" };
+      "mode[4]"      string => "enforce";
+      "status[4]"    string => "repaired";
```
Comment (ncharles) [inline tests/acceptance/30_generic_methods/permissions_user_acl_present.cf]: ok, you'll have to explain that one: it's already -rwx, so if we -rw, nothing changed

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**